api/src/main/java/org/openmrs/module/cohort/api/impl/CohortTypeServiceImpl.java

```diff
@@ -59,8 +58,7 @@ public void voidCohortType(@NotNull CohortType cohortType, String voidedReason)
 		if (cohortType == null) {
 			return;
 		}
-		
-		Context.getService(CohortTypeService.class).saveCohortType(cohortType);
+		dao.createOrUpdate(cohortType);
```

Comment:
I tested this locally and found that here (in cohortTypes), it voids the cohortType automatically even if we haven't manually set the fields related to voiding. 

But in cohorts, it doesn't automatically void the cohort unless we manually set the fields related to voiding.

Reply:
Theoretically, there's a VoidHandler AOP that's supposed to do this for most cases... Why it isn't working for Cohorts is a good question, though...

Reply:
For the voiding to happen automatically, the method should be renamed to `voidCohortM` because of this: https://github.com/openmrs/openmrs-core/blob/2.6.0/api/src/main/java/org/openmrs/aop/RequiredDataAdvice.java#L191-L218